### DISCUSSION TITLE
spaCy VSCode: Error selecting python interpreter

### DISCUSSION THREAD

**Discussion by adamkgoldfarb** [explosion/spaCy#12673] [Help: Installation] [answered]
Hello! I use spacy with a miniconda environment and am looking forward to trying out the VSCode Extension. When I select the `~/miniconda3/envs/myenv` directory I receive a "Selected python interpreter not compatible" error, and "Select current interpreter" does not work either. Any tips for how to debug this issue? I opened the developer console but could not identify an error message.

**Comment by victorialslocum** [comment, marked as answer]
Hi! Do you have the two dependencies (spaCy >= 3.4.0 & pygls >= 1.0.0) installed in the environment?

**Reply by adamkgoldfarb** [reply]
Thanks-- I had spaCy 3.5.3 installed but not pygls! That did it, and auto-selecting the interpreter works. Appreciate the quick help!

**Reply by adamkgoldfarb** [reply]
Hello! Following up on this as pygls==1.1.1 and spacy==3.6.1 seem to have a conflict-- I'm getting the error from earlier in the thread w/r/t not being able to auto-load or select an interpreter. Any steps to debug the issue would be welcome!

**Reply by shadeMe** [reply]
Thanks for the report! We'll investigate. In the meantime, could you please post the modules installed in your venv, i.e., the output of `pip list` (or `conda list` if you use miniconda)?

**Reply by shadeMe** [reply]
It seems like an upstream change introduced in `pygls 1.1` has changed an API that we were using to query its version. We'll look into a fix for this issue. 

Until then, `pygls` can be reverted to `1.0.0` to workaround this issue.

**Reply by adamkgoldfarb** [reply]
Thank you! Sorry I missed the request for the conda freeze request-- it's below.

Name                    Version                   Build  Channel
aiofiles                  23.2.1                   pypi_0    pypi
aiohttp                   3.8.6                    pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
altair                    5.1.2                    pypi_0    pypi
annotated-types           0.6.0                    pypi_0    pypi
anyio                     4.0.0              pyhd8ed1ab_0    conda-forge
appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
argon2-cffi               23.1.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py311heffc1b2_4    conda-forge
arrow                     1.3.0              pyhd8ed1ab_0    conda-forge
asttokens                 2.4.0              pyhd8ed1ab_0    conda-forge
async-lru                 2.0.4              pyhd8ed1ab_0    conda-forge
async-timeout             4.0.3                    pypi_0    pypi
atk-1.0                   2.38.0               hcb7b3dd_1    conda-forge
attrs                     23.1.0             pyh71513ae_1    conda-forge
babel                     2.13.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
bleach                    6.1.0              pyhd8ed1ab_0    conda-forge
blis                      0.7.11                   pypi_0    pypi
bottleneck                1.3.7                    pypi_0    pypi
brotli                    1.1.0                hb547adb_1    conda-forge
brotli-bin                1.1.0                hb547adb_1    conda-forge
brotli-python             1.1.0           py311ha891d26_1    conda-forge
bzip2                     1.0.8                h3422bc3_4    conda-forge
ca-certificates           2023.7.22            hf0a4a13_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cachetools                5.3.1                    pypi_0    pypi
cairo                     1.18.0               hd1e100b_0    conda-forge
catalogue                 2.0.10                   pypi_0    pypi
cattrs                    23.1.2                   pypi_0    pypi
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py311h4a08483_0    conda-forge
charset-normalizer        3.3.0              pyhd8ed1ab_0    conda-forge
click                     8.1.7                    pypi_0    pypi
cloudpathlib              0.16.0                   pypi_0    pypi
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
comm                      0.1.4              pyhd8ed1ab_0    conda-forge
confection                0.1.3                    pypi_0    pypi
contourpy                 1.1.1           py311he4fd1f5_1    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
cymem                     2.0.8                    pypi_0    pypi
debugpy                   1.8.0           py311ha891d26_1    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
dnspython                 2.4.2                    pypi_0    pypi
einops                    0.7.0                    pypi_0    pypi
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
et-xmlfile                1.1.0                    pypi_0    pypi
exceptiongroup            1.1.3              pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
expat                     2.5.0                hb7217d7_1    conda-forge
faiss-cpu                 1.7.4                    pypi_0    pypi
fastapi                   0.102.0                  pypi_0    pypi
fasttext                  0.9.2                    pypi_0    pypi
filelock                  3.12.4             pyhd8ed1ab_0    conda-forge
floret                    0.10.4                   pypi_0    pypi
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.14.2               h82840c6_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.43.1          py311h05b510d_0    conda-forge
fqdn                      1.5.1              pyhd8ed1ab_0    conda-forge
freetype                  2.12.1               hadb7bae_2    conda-forge
fribidi                   1.0.10               h27ca646_0    conda-forge
frozenlist                1.4.0                    pypi_0    pypi
fsspec                    2023.9.2                 pypi_0    pypi
gdk-pixbuf                2.42.10              h15fa40c_4    conda-forge
gettext                   0.21.1               h0186832_0    conda-forge
giflib                    5.2.1                h1a8c8d9_3    conda-forge
gmp                       6.2.1                h9f76cd9_0    conda-forge
gmpy2                     2.1.2           py311h2ba9262_1    conda-forge
graphite2                 1.3.13            h9f76cd9_1001    conda-forge
graphviz                  8.1.0                h10878c0_0    conda-forge
gtk2                      2.24.33              h57013de_2    conda-forge
gts                       0.7.6                he42f4ea_4    conda-forge
h11                       0.14.0             pyhd8ed1ab_0    conda-forge
harfbuzz                  8.2.1                hf1a6348_0    conda-forge
html5lib                  1.1                      pypi_0    pypi
huggingface-hub           0.18.0                   pypi_0    pypi
icu                       73.2                 hc8870d7_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
importlib-metadata        6.8.0              pyha770c72_0    conda-forge
importlib_metadata        6.8.0                hd8ed1ab_0    conda-forge
importlib_resources       6.1.0              pyhd8ed1ab_0    conda-forge
inflection                0.5.1                    pypi_0    pypi
ipykernel                 6.25.2             pyh1050b4e_0    conda-forge
ipython                   8.16.1             pyh31c8845_0    conda-forge
ipywidgets                8.1.1              pyhd8ed1ab_0    conda-forge
isoduration               20.11.0            pyhd8ed1ab_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
joblib                    1.3.2              pyhd8ed1ab_0    conda-forge
json5                     0.9.14             pyhd8ed1ab_0    conda-forge
jsonpointer               2.4             py311h267d04e_3    conda-forge
jsonschema                4.19.1             pyhd8ed1ab_0    conda-forge
jsonschema-specifications 2023.7.1           pyhd8ed1ab_0    conda-forge
jsonschema-with-format-nongpl 4.19.1             pyhd8ed1ab_0    conda-forge
jupyter-lsp               2.2.0              pyhd8ed1ab_0    conda-forge
jupyter_client            8.4.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.4.0           py311h267d04e_0    conda-forge
jupyter_events            0.7.0              pyhd8ed1ab_2    conda-forge
jupyter_server            2.7.3              pyhd8ed1ab_1    conda-forge
jupyter_server_terminals  0.4.4              pyhd8ed1ab_1    conda-forge
jupyterlab                4.0.7              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
jupyterlab_server         2.25.0             pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        3.0.9              pyhd8ed1ab_0    conda-forge
kiwisolver                1.4.5           py311he4fd1f5_1    conda-forge
krb5                      1.21.2               h92f50d5_0    conda-forge
langcodes                 3.3.0                    pypi_0    pypi
lcms2                     2.15                 hf2736f0_3    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libabseil                 20230802.1      cxx17_h13dd4ca_0    conda-forge
libblas                   3.9.0           18_osxarm64_openblas    conda-forge
libbrotlicommon           1.1.0                hb547adb_1    conda-forge
libbrotlidec              1.1.0                hb547adb_1    conda-forge
libbrotlienc              1.1.0                hb547adb_1    conda-forge
libcblas                  3.9.0           18_osxarm64_openblas    conda-forge
libcxx                    16.0.6               h4653b0c_0    conda-forge
libdeflate                1.19                 hb547adb_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libexpat                  2.5.0                hb7217d7_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgd                     2.3.3                hfdf3952_9    conda-forge
libgfortran               5.0.0           13_2_0_hd922786_1    conda-forge
libgfortran5              13.2.0               hf226fd6_1    conda-forge
libglib                   2.78.0               h24e9cb9_0    conda-forge
libiconv                  1.17                 he4db4b2_0    conda-forge
libjpeg-turbo             3.0.0                hb547adb_1    conda-forge
liblapack                 3.9.0           18_osxarm64_openblas    conda-forge
libopenblas               0.3.24          openmp_hd76b1f2_0    conda-forge
libpng                    1.6.39               h76d750c_0    conda-forge
libpq                     16.0                 hcea71ed_1    conda-forge
libprotobuf               4.24.3               hf590ac1_1    conda-forge
librsvg                   2.56.3               h0db3404_0    conda-forge
libsentencepiece          0.1.99               hdd34817_4    conda-forge
libsodium                 1.0.18               h27ca646_1    conda-forge
libsqlite                 3.43.2               h091b4b1_0    conda-forge
libtiff                   4.6.0                ha8a6c65_2    conda-forge
libtool                   2.4.7                hb7217d7_0    conda-forge
libwebp                   1.3.2                hf30222e_1    conda-forge
libwebp-base              1.3.2                hb547adb_0    conda-forge
libxcb                    1.15                 hf346824_0    conda-forge
libxml2                   2.11.5               h25269f3_1    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
llvm-openmp               17.0.2               h1c12783_0    conda-forge
llvmlite                  0.41.0                   pypi_0    pypi
loguru                    0.7.2                    pypi_0    pypi
lsprotocol                2023.0.0b1               pypi_0    pypi
lxml                      4.9.3                    pypi_0    pypi
markupsafe                2.1.3           py311heffc1b2_1    conda-forge
matplotlib-base           3.8.0           py311hfdba5f6_2    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mistune                   3.0.1              pyhd8ed1ab_0    conda-forge
mpc                       1.3.1                h91ba8db_0    conda-forge
mpfr                      4.2.0                he09a6ba_0    conda-forge
mpmath                    1.3.0              pyhd8ed1ab_0    conda-forge
multidict                 6.0.4                    pypi_0    pypi
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
murmurhash                1.0.10                   pypi_0    pypi
nbclient                  0.8.0              pyhd8ed1ab_0    conda-forge
nbconvert-core            7.9.2              pyhd8ed1ab_0    conda-forge
nbformat                  5.9.2              pyhd8ed1ab_0    conda-forge
ncurses                   6.4                  h7ea286d_0    conda-forge
nest-asyncio              1.5.8              pyhd8ed1ab_0    conda-forge
networkx                  3.1                pyhd8ed1ab_0    conda-forge
notebook-shim             0.2.3              pyhd8ed1ab_0    conda-forge
numba                     0.58.0                   pypi_0    pypi
numexpr                   2.8.7                    pypi_0    pypi
numpy                     1.25.2                   pypi_0    pypi
odfpy                     1.4.1                    pypi_0    pypi
openai                    0.28.1                   pypi_0    pypi
openjpeg                  2.5.0                h4c1507b_3    conda-forge
openpyxl                  3.1.2                    pypi_0    pypi
openssl                   3.1.3                h53f4e23_0    conda-forge
outcome                   1.2.0              pyhd8ed1ab_0    conda-forge
overrides                 7.4.0              pyhd8ed1ab_0    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pandas                    2.1.1           py311h9e438b8_1    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
pango                     1.50.14              hcf40dda_2    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pathy                     0.10.2                   pypi_0    pypi
patsy                     0.5.3              pyhd8ed1ab_0    conda-forge
pcre2                     10.40                hb34f9b4_0    conda-forge
peewee                    3.16.3                   pypi_0    pypi
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.1.0          py311hb9c5795_0    conda-forge
pinecone-client           2.2.4                    pypi_0    pypi
pip                       23.3               pyhd8ed1ab_0    conda-forge
pixman                    0.42.2               h13dd4ca_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_1    conda-forge
platformdirs              3.11.0             pyhd8ed1ab_0    conda-forge
plotext                   5.2.8                    pypi_0    pypi
plotly                    5.17.0                     py_0    plotly
preshed                   3.0.9                    pypi_0    pypi
prodigy                   1.14.4                   pypi_0    pypi
prometheus_client         0.17.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.39             pyha770c72_0    conda-forge
prompt_toolkit            3.0.39               hd8ed1ab_0    conda-forge
protobuf                  4.24.3          py311h4d1eceb_1    conda-forge
psutil                    5.9.5           py311heffc1b2_1    conda-forge
psycopg2                  2.9.7           py311h589e011_1    conda-forge
psycopg2-binary           2.9.7              pyhd8ed1ab_1    conda-forge
pthread-stubs             0.4               h27ca646_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyairtable                2.1.0.post1              pypi_0    pypi
pyarrow                   13.0.0                   pypi_0    pypi
pybind11                  2.11.1                   pypi_0    pypi
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydantic                  2.4.2                    pypi_0    pypi
pydantic-core             2.10.1                   pypi_0    pypi
pygls                     1.1.1                    pypi_0    pypi
pygments                  2.16.1             pyhd8ed1ab_0    conda-forge
pygraphviz                1.11            py311h5178850_1    conda-forge
pyjwt                     2.8.0                    pypi_0    pypi
pymsteams                 0.2.2                    pypi_0    pypi
pyobjc-core               10.0            py311hb702dc4_0    conda-forge
pyobjc-framework-cocoa    10.0            py311hb702dc4_1    conda-forge
pyparsing                 3.1.1              pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.11.6          h47c9636_0_cpython    conda-forge
python-chromedriver-binary 2.42.0                     py_0    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-dotenv             1.0.0              pyhd8ed1ab_1    conda-forge
python-fastjsonschema     2.18.1             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
python-tzdata             2023.3             pyhd8ed1ab_0    conda-forge
python_abi                3.11                    4_cp311    conda-forge
pytorch                   2.1.0                  py3.11_0    pytorch
pytz                      2023.3.post1       pyhd8ed1ab_0    conda-forge
pyxlsb                    1.0.10                   pypi_0    pypi
pyyaml                    6.0.1           py311heffc1b2_1    conda-forge
pyzmq                     25.1.1          py311hb1af645_1    conda-forge
radicli                   0.0.25                   pypi_0    pypi
readline                  8.2                  h92ec313_1    conda-forge
referencing               0.30.2             pyhd8ed1ab_0    conda-forge
regex                     2023.10.3                pypi_0    pypi
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
rpds-py                   0.10.6          py311h94f323b_0    conda-forge
safetensors               0.4.0                    pypi_0    pypi
scikit-learn              1.3.1           py311ha25ca4d_1    conda-forge
scipy                     1.11.3          py311h93d07a4_1    conda-forge
seaborn                   0.13.0               hd8ed1ab_0    conda-forge
seaborn-base              0.13.0             pyhd8ed1ab_0    conda-forge
selenium                  4.13.0             pyhd8ed1ab_0    conda-forge
selenium-manager          4.13.0               h69fbcac_0    conda-forge
send2trash                1.8.2              pyhd1c38e8_0    conda-forge
sentencepiece             0.1.99               ha1ab1f8_4    conda-forge
sentencepiece-python      0.1.99          py311h2fa77fb_4    conda-forge
sentencepiece-spm         0.1.99               hdd34817_4    conda-forge
setuptools                68.2.2             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smart-open                6.4.0                    pypi_0    pypi
sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.5                pyhd8ed1ab_1    conda-forge
spacy                     3.6.1                    pypi_0    pypi
spacy-alignments          0.9.1                    pypi_0    pypi
spacy-legacy              3.0.12                   pypi_0    pypi
spacy-llm                 0.5.1                    pypi_0    pypi
spacy-loggers             1.0.5                    pypi_0    pypi
spacy-lookups-data        1.0.5                    pypi_0    pypi
spacy-transformers        1.2.5                    pypi_0    pypi
srsly                     2.4.8                    pypi_0    pypi
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
starlette                 0.27.0                   pypi_0    pypi
statsmodels               0.14.0          py311h9ea6feb_2    conda-forge
sympy                     1.12            pypyh9d50eac_103    conda-forge
tabulate                  0.9.0                    pypi_0    pypi
tenacity                  8.2.3              pyhd8ed1ab_0    conda-forge
terminado                 0.17.1             pyhd1c38e8_0    conda-forge
text-unidecode            1.3                pyhd8ed1ab_1    conda-forge
thinc                     8.1.12                   pypi_0    pypi
thinc-apple-ops           0.1.4                    pypi_0    pypi
threadpoolctl             3.2.0              pyha21a80b_0    conda-forge
tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.13               hb31c410_0    conda-forge
tokenizers                0.13.3                   pypi_0    pypi
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
toolz                     0.12.0                   pypi_0    pypi
tornado                   6.3.3           py311heffc1b2_1    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
traitlets                 5.11.2             pyhd8ed1ab_0    conda-forge
transformers              4.30.2                   pypi_0    pypi
trio                      0.22.2          py311h267d04e_1    conda-forge
trio-websocket            0.11.1             pyhd8ed1ab_0    conda-forge
typeguard                 3.0.2                    pypi_0    pypi
typer                     0.9.0                    pypi_0    pypi
types-python-dateutil     2.8.19.14          pyhd8ed1ab_0    conda-forge
typing-extensions         4.8.0                hd8ed1ab_0    conda-forge
typing_extensions         4.8.0              pyha770c72_0    conda-forge
typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
uri-template              1.3.0              pyhd8ed1ab_0    conda-forge
urllib3                   2.0.6              pyhd8ed1ab_0    conda-forge
uvicorn                   0.18.3                   pypi_0    pypi
wasabi                    1.1.2                    pypi_0    pypi
wcwidth                   0.2.8              pyhd8ed1ab_0    conda-forge
webcolors                 1.13               pyhd8ed1ab_0    conda-forge
webdriver-manager         4.0.1              pyhd8ed1ab_0    conda-forge
webencodings              0.5.1              pyhd8ed1ab_2    conda-forge
websocket-client          1.6.4              pyhd8ed1ab_0    conda-forge
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
widgetsnbextension        4.0.9              pyhd8ed1ab_0    conda-forge
wsproto                   1.2.0              pyhd8ed1ab_0    conda-forge
xlrd                      2.0.1                    pypi_0    pypi
xlsxwriter                3.1.8                    pypi_0    pypi
xorg-libxau               1.0.11               hb547adb_0    conda-forge
xorg-libxdmcp             1.1.3                h27ca646_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
yaml                      0.2.5                h3422bc3_2    conda-forge
yarl                      1.9.2                    pypi_0    pypi
zeromq                    4.3.5                h965bd2d_0    conda-forge
zipp                      3.17.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h53f4e23_5    conda-forge
zstd                      1.5.5                h4f39d0f_0    conda-forge
